packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -149,8 +149,13 @@ abstract class XcodeBasedProject extends FlutterProjectPlatform {
   /// Checks if FlutterGeneratedPluginSwiftPackage has been added to the
   /// project's build settings by checking the contents of the pbxproj.
   bool get flutterPluginSwiftPackageInProjectSettings {
-    return xcodeProjectInfoFile.existsSync() &&
-        xcodeProjectInfoFile.readAsStringSync().contains('FlutterGeneratedPluginSwiftPackage');
+    if (!xcodeProjectInfoFile.existsSync()) {
```

Comment:
The old behavior was a bit conservative, which prevented the migrator from running the new check for the custom targets.

Updated all the relevant implementations of this getter to match the new behavior.